src/algebra/hom/iterate.lean

```diff
@@ -164,8 +164,30 @@ smul_iterate (mul_opposite.op a) n
 lemma mul_right_iterate_apply_one : (* a)^[n] 1 = a ^ n :=
 by simp [mul_right_iterate]
 
+@[simp, to_additive]
+lemma pow_iterate (n : ℕ) (j : ℕ) : ((λ (x : G), x^n)^[j]) = λ x, x^(n^j) :=
+begin
+  induction j with j ih, { simpa, },
+  ext,
+  simp only [iterate_succ, comp_app, ih, ← pow_mul, pow_succ],
+end
+
```

Comment:
Does `attribute [to_additive] smul_iterate` do something sensible after this line?